test/app_test.rb

```diff
@@ -0,0 +1,35 @@
+class TestApp < Test::Unit::TestCase
+  def setup
+    @app = App.new
+    @book = Book.new('The Alchemist', 'Paulo Coelho')
+    @person = Person.new(30, 'John', 1)
+    @rental = Rental.new('2022-07-06', @book, @person)
+  end
```

Comment:
There are a couple of issues with the way you have implemented this milestone requirement:

- The folder containing the test should be called spec and not test in order to follow rspec naming convention
- The test files should be named like this `filename_spec.rb` to enable them visible to rspec and to follow their naming conventions
- Also the way your test is written doesn't seem to utilize rspec, I recommend following this [tutorial](https://semaphoreci.com/community/tutorials/getting-started-with-rspec) to get yourselves started

In order to stick with the requirements, consider observing the above.

HINT: When you're done with the tests, your test files should look similar to the below snippet

```ruby
describe Person do
  context "When creating a new instance of the Person class" do
    it "The constructor method returns a Person object with provided age, name, and empty rentals" do
      age = 27
      name = "Matt Smith"
      person = Person.new(age, name, false)

      expect(person.age).to eql age
      expect(person.name).to eql name
      expect(person.rentals).to eql []
    end

    it "The constructor method sets default value for the optional name parameter if it is not provided" do
      person = Person.new(age)
      expect(person.name).to eql "Unknown"
    end
  end
```

Reply:
Hi @DuaneDave 
I am so happy to your feedback.
We did using assertion test method.
Example :app_spec.test
```
require "test/unit"
require_relative "../app"
class TestApp < Test::Unit::TestCase
  def setup
    @app = App.new
    @book = Book.new("The Alchemist", "Paulo Coelho")
    @person = Person.new(30, "John", 1)
    @rental = Rental.new("2022-07-06", @book, @person)
  end

  def test_addperson
    @app.addperson(@person)
    assert_equal([@person], @app.getpersons)
  end

  def test_getpersons
    assert_equal([], @app.getpersons)
  end

  def test_addbook
    @app.addbook(@book)
    assert_equal([@book], @app.getbooks)
  end

  def test_getbooks
    assert_equal([], @app.getbooks)
  end

  def test_addrental
    @app.addrental(@rental)
    assert_equal([@rental], @app.getrentals)
  end

  def test_getrentals
    assert_equal([], @app.getrentals)
  end
end
```
- both **class** that extend Unit test is the same describe
- setup is also similar to **before(:context) ** to initialize your class variable
### We follow the following Microverse guide line 
### Unit testing
You already know what unit testing is from the previous modules, so let's just refresh our knowledge about them by reading the following article:

- [Unit testing best practices](https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices)

### Rspec
Rspec is a framework composed of multiple libraries designed to work together to make for the best possible experience when making tests.

To learn more about Rspec, read the following articles:
- [Rspec tutorial: Quick guide](https://www.tutorialspoint.com/rspec/rspec_quick_guide.htm)
- [Testing with Rspec](https://code.tutsplus.com/tutorials/ruby-for-newbies-testing-with-rspec--net-21297)
- [Using Rspec to build a calculator](https://medium.com/swlh/basic-testing-in-ruby-using-rspec-to-build-a-simple-calculator-e747d53cc0f7)

------